tendrl/node_agent/ini2json.py

```diff
@@ -0,0 +1,125 @@
+                # no section header in the file?
+                elif cursect is None:
+                    raise MissingSectionHeaderError(fpname, lineno, line)
+                # an option line?
+                else:
+                    try:
+                        mo = self._optcre.match(line)   # 2.7
+                    except AttributeError:
+                        mo = self.OPTCRE.match(line)    # 2.6
+                    if mo:
+                        optname, vi, optval = mo.group('option', 'vi', 'value')
+                        optname = self.optionxform(optname.rstrip())
+                        # This check is fine because the OPTCRE cannot
+                        # match if it would set optval to None
+                        if optval is not None:
+                            if vi in ('=', ':') and ';' in optval:
+                                # ';' is a comment delimiter only if it follows
+                                # a spacing character
+                                pos = optval.find(';')
+                                if pos != -1 and optval[pos - 1].isspace():
+                                    optval = optval[:pos]
+                            optval = optval.strip()
+                            # allow empty values
+                            if optval == '""':
+                                optval = ''
+                            cursect[optname] = [optval]
+                        else:
+                            # valueless option handling
+                            cursect[optname] = optval
+                    else:
+                        # a non-fatal parsing error occurred.  set up the
+                        # exception but keep going. the exception will be
+                        # raised at the end of the file and will contain a
+                        # list of all bogus lines
+                        if not e:
+                            e = ParsingError(fpname)
+                        e.append(lineno, repr(line))
+        # if any parsing errors occurred, raise an exception
+        if e:
+            raise e
+
+        # join the multi-line values collected while reading
+        all_sections = [self._defaults]
+        all_sections.extend(self._sections.values())
+        for options in all_sections:
+            for name, val in options.items():
+                if isinstance(val, list):
+                    options[name] = '\n'.join(val)
+
+    def dget(self, section, option, default=None, type=str):
+        if not self.has_option(section, option):
+            return default
+        if type is str:
+            return self.get(section, option)
+        elif type is int:
+            return self.getint(section, option)
+        elif type is bool:
+            return self.getboolean(section, option)
+        else:
+            raise NotImplementedError()
+
+
+def ini_to_dict(ini_file_path):
+    cfg = StrictConfigParser()
+    f = open(ini_file_path)
```

Comment:
@brainfunked this piece of code is simply copied from gluster-integration. Later we plan to move this code to commons module and we can refine it that time.